mqt/qfr/qiskit/QuantumCircuit.hpp

```diff
@@ -192,6 +197,77 @@ namespace qc::qiskit {
             }
         }
 
+        static SymbolOrNumber parseSymbolicExpr(const py::object& pyExpr) {
+            static const std::regex summands("[+|-]?[^+-]+");
+            static const std::regex products("[\\*/]?[^\\*/]+");
+            // static const std::regex products(".*");
+            // static const products("")
+            auto exprStr = pyExpr.attr("__str__")().cast<std::string>();
+            exprStr.erase(std::remove(exprStr.begin(), exprStr.end(), ' '),
+                          exprStr.end()); // strip whitespace
+
+            auto sumIt  = std::sregex_iterator(exprStr.begin(), exprStr.end(), summands);
+            auto sumEnd = std::sregex_iterator();
+
+            qc::Symbolic sym;
+            bool         isConst = true;
+
+            while (sumIt != sumEnd) {
+                auto match    = *sumIt;
+                auto matchStr = match.str();
+                int  sign     = matchStr[0] == '-' ? -1 : 1;
+                if (matchStr[0] == '+' || matchStr[0] == '-') {
+                    matchStr.erase(0, 1);
+                }
+                // std::cout << matchStr << std::endl;
+
+                auto prodIt  = std::sregex_iterator(matchStr.begin(), matchStr.end(), products);
+                auto prodEnd = std::sregex_iterator();
+
+                dd::fp      coeff = 1.0;
+                std::string var   = "";
+                while (prodIt != prodEnd) {
+                    auto prodMatch = *prodIt;
+                    auto prodStr   = prodMatch.str();
+
+                    bool isDiv = prodStr[0] == '/';
+                    if (prodStr[0] == '*' || prodStr[0] == '/') {
+                        prodStr.erase(0, 1);
+                    }
+
+                    std::istringstream iss(prodStr);
+                    dd::fp             f;
+                    iss >> f;
+                    bool isNum = iss.eof() && !iss.fail();
+
+                    if (isNum) {
```

Comment:
```suggestion
                    if (iss.eof() && !iss.fail()) {
```

Reply:
That is embarrassing. That is freshmen-level coding.